test/TesterInternal/SchedulerTests/OrleansTaskSchedulerAdvancedTests_Set2.cs

```diff
@@ -604,19 +604,32 @@ public void ActivationSched_SubTaskExecutionSequencing()
             {
                 TimeSpan waitCheckTime = TimeSpan.FromMilliseconds(150 * ChainLength * NumChains * waitFactor);
 
-                var timeoutLimit = TimeSpan.FromSeconds(1);
                 try
                 {
                     await resultHandles[i].Task.WithTimeout(waitCheckTime);
                 }
                 catch (TimeoutException)
                 {
-                    Assert.True(false, "Result did not arrive before timeout " + timeoutLimit);
+                    Assert.True(false, "Result did not arrive before timeout " + waitCheckTime);
                 }
 
                 bool ok = resultHandles[i].Task.Result;
-                Assert.False(taskChainEnds[i].IsFaulted, "Task chain " + i + " should not be Faulted: " + taskChainEnds[i].Exception);
+                
+                if (!taskChainEnds[i].IsCompleted)
+                {
+                    try
+                    {
+                        // since resultHandle being complete doesn't directly imply that the final chain was completed (there's a chance for a race condition), give a small chance for it to complete.
+                        await taskChainEnds[i].WithTimeout(TimeSpan.FromMilliseconds(10));
```

Comment:
didn't want to do this, as I wasn't sure about the intentions... otherwise the more correct way would be to get rid of the resultHandle stuff, and just await the chain end...